### DISCUSSION TITLE
Change HL7 mapping file

### DISCUSSION THREAD

**Discussion by jbravo94** [dcm4che/dcm4chee-arc-light#3407] [Q&A] [answered]
Dear all,

we installed dcm4chee arc successfully and MWL works basically with some HL7 messages.

We want to adapt the HL7 ORM to DICOM Tag mapping which should be possible via hl7-order2dcm.xsl

According to our researched we should be able to use this custommapping via the option located under

https://github.com/dcm4che/dcm4chee-arc-light/wiki/HL7-Receivers-feeding-MWL-SCPs#hl7-to-mwl-mapping

We found this option
![image](https://user-images.githubusercontent.com/26746017/136403497-3807d2f8-3b67-43ab-b9ff-c0f23499eae8.png)

where we tried to insert
/opt/wildfly/standalone/hl7-order2dcm.xsl
or
file:///opt/wildfly/standalone/hl7-order2dcm.xsl

and this option
![image](https://user-images.githubusercontent.com/26746017/136403718-dc3f216d-0f7b-4955-a8f7-c84db936da93.png)
/opt/wildfly/standalone/hl7-common.xsl
or
file:///opt/wildfly/standalone/hl7-common.xsl

wheras the files were accessible - but it did not work unfortunately...

We kindly ask for a hint.

P.S.: We intentionally do not use an integration engine and change the mapping of the hl7 via conformance statement

BR,
Johannes

**Comment by vrindanayak** [comment, marked as answer]
`hl7-common.xsl` is a common stylesheet referenced from other stylesheets used for patient / order / report management services. Specifying `/opt/wildfly/standalone/hl7-common.xsl` or `file:///opt/wildfly/standalone/hl7-common.xsl` is therefore incorrect.

By default, 
- `HL7 Schedule Procedure Template URI` is configured to `${jboss.server.temp.url}/dcm4chee-arc/hl7-order2dcm.xsl` 
  and 
- `HL7 Patient Update Template URI` is configured to `${jboss.server.temp.url}/dcm4chee-arc/hl7-adt2dcm.xsl`. 

![image](https://user-images.githubusercontent.com/15785031/136918634-f2189efe-3bac-41a3-a292-92221a8fd039.png)
 
These stylesheets are part of vendor data which can be accessed from device (`Configuration` page -> `Devices` tab -> `dcm4chee-arc` device -> `Attributes` -> `Vendor Device Data`). 

![image](https://user-images.githubusercontent.com/15785031/136918349-ab1e9472-3bce-4f5e-8cdb-55de0f98d886.png)


You can customize existing stylesheets by updating the vendor data (Download (existing) -> Delete -> Upload (updated)). See [Vendor Data](https://github.com/dcm4che/dcm4chee-arc-light/wiki/Vendor-Data) wiki
